src/coreclr/vm/assembly.cpp

```diff
@@ -225,9 +231,18 @@ Assembly::~Assembly()
 
     if (m_pPEAssembly)
     {
+        if (m_fHostAssemblyPublished)
```

Comment:
Do we need this flag? We can clear the host assembly backlink unconditionally, even when it was not set yet.